alacritty_terminal/src/window.rs

```diff
@@ -409,6 +421,49 @@ impl Window {
     }
 }
 
+#[cfg(not(any(target_os = "macos", windows)))]
+fn x_embed_window(window: &GlutinWindow, parent_window_id: u64) {
+    let (xlib_display, xlib_window) = match (window.get_xlib_display(), window.get_xlib_window()) {
+        (Some(display), Some(window)) => (display, window),
+        _ => return,
+    };
+
+    let xlib = Xlib::open().expect("get xlib");
+
+    unsafe {
+        let xembed = "_XEMBED".as_ptr();
+        let atom = (xlib.XInternAtom)(xlib_display as *mut _, xembed as *const _, 0);
+        (xlib.XChangeProperty)(
+            xlib_display as _,
+            xlib_window as _,
+            atom,
+            atom,
+            32,
+            PropModeReplace,
+            [0, 1].as_ptr(),
+            2,
+        );
+        
+        // set a handler to deal with unsuccessful reparenting
+        let old_handler = (xlib.XSetErrorHandler)(Some(ctx_error_handler));
+        // reparent the window with a specific handler in-place
+        (xlib.XReparentWindow)(xlib_display as _, xlib_window as _, parent_window_id, 0, 0);
+        // XReparentWindow doesn't give helpful results, so poke the window here
+        let mut attribs: x11_dl::xlib::XWindowAttributes = std::mem::uninitialized();
+        (xlib.XGetWindowAttributes)(xlib_display as _, xlib_window as _, &mut attribs);
+        // restore the original handler
+        (xlib.XSetErrorHandler)(std::mem::transmute(old_handler));
+    }
+}
+
+unsafe extern "C" fn ctx_error_handler(
+        _dpy: *mut x11_dl::xlib::Display,
+        _ev: *mut x11_dl::xlib::XErrorEvent
+        ) -> i32 {
+    eprintln!("Could not embed into specified window.");
+    ::std::process::exit(1);
```

Comment:
It's quite old syntax here, please, convert it to `process::exit(1)` + import at top.

Reply:
Done.